### PR TITLE
remove redundant pytest-twisted

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by hynek** [comment]
Would you mind adding some comments about what it does and how it does it? To me this looks like gibberish that accidentally passes all tests. :D

File: tests/test_tx.py [hunk 1]
```diff
@@ -14,18 +14,84 @@
...
 import pytest
-import pytest_twisted
 
-from twisted.internet.defer import Deferred, fail, succeed
+from twisted.internet.defer import Deferred, Failure, fail, succeed
 
 from prometheus_async import tx
 
 
+def _from_async_fn(async_fn):
```
Comment (graingert) [inline tests/test_tx.py:26]:
```suggestion
def _from_async_fn(async_fn):
    # this code is based on
    # https://docs.twisted.org/en/twisted-22.8.0/api/twisted.trial._synctest._Assertions.html#successResultOf
    # except it takes a coroutine function and wraps it in a Deferred first
```

File: tests/test_tx.py [hunk 2]
```diff
@@ -14,18 +14,87 @@
...
+    # https://docs.twisted.org/en/twisted-22.8.0/api/twisted.trial._synctest._Assertions.html#successResultOf
+    # except it takes a coroutine function and wraps it in a Deferred first
+    @functools.wraps(async_fn)
+    def wrapper(*args, **kwargs):
+        results = []
+        d = Deferred.fromCoroutine(async_fn(*args, **kwargs)).addBoth(
+            results.append
+        )
+        try:
+            if not results:
```
Comment (graingert) [inline tests/test_tx.py:37]:
```suggestion
            if not results:
                # none of the tests here use the reactor and so Deferred should always
                # immediately complete and this branch will never execute
```

File: tests/test_tx.py [hunk 3]
```diff
@@ -14,18 +14,89 @@
...
+                # none of the tests here use the reactor and so Deferred should always
```
Comment (graingert) [inline tests/test_tx.py:38]:
```suggestion
                # none of the tests here use the reactor and so
                # the Deferred should always
```

File: tests/test_tx.py [hunk 4]
```diff
@@ -14,18 +14,90 @@
...
+    @functools.wraps(async_fn)
+    def wrapper(*args, **kwargs):
+        results = []
+        d = Deferred.fromCoroutine(async_fn(*args, **kwargs)).addBoth(
+            results.append
+        )
+        try:
+            if not results:
+                # none of the tests here use the reactor and so
+                # the Deferred should always
```
Comment (graingert) [inline tests/test_tx.py:39]:
```suggestion
                # the Deferred should always immediately complete
```

File: tests/test_tx.py [hunk 5]
```diff
@@ -14,18 +14,90 @@
...
+                # immediately complete and this branch will never execute
```
Comment (graingert) [inline tests/test_tx.py:40]:
```suggestion
                # and so this branch will never execute
```

File: tests/test_tx.py [hunk 6]
```diff
@@ -14,18 +14,90 @@
...
+class TestFromAsyncFn:
+    def test_no_result(self):
+        @_from_async_fn
+        async def demo():
+            return await Deferred()
+
+        with pytest.raises(
+            RuntimeError,
+            match=r"Success result expected on <Deferred at 0x.*>, "
+            "found no result instead",
```
Comment (graingert) [inline tests/test_tx.py:68]:
```suggestion
            match=r"Success result expected on <Deferred at 0x.*>, " "found no result instead",
```

**Review by hynek — APPROVED** [review]
Thanks! Bear in mind that I'll @ you whenever any of this breaks now. :)

**Comment by graingert** [comment]
> Thanks! Bear in mind that I'll @ you whenever any of this breaks now. :)

of course! feel free to ping me in my discord too!

@hynek can you also set https://github.com/hynek/prometheus-async/settings/actions#actions_fork_pr_approvals_first_time_contributor_new_users ?

**Pull request merged**